### ISSUE TITLE
INTERNAL ERROR

### ISSUE THREAD

**Issue by kvz** [filiph/linkcheck#51]
Doing as instructed:

```bash
Sorry! Please open https://github.com/filiph/linkcheck/issues/new in your favorite browser and copy paste the following output there:

Bad state: No element
#0      SetMixin.singleWhere (dart:collection/set.dart:271)
#1      crawl.<anonymous closure> (package:linkcheck/src/crawl.dart:255)
#2      _rootRunUnary (dart:async/zone.dart:1132)
#3      _CustomZone.runUnary (dart:async/zone.dart:1029)
#4      _CustomZone.runUnaryGuarded (dart:async/zone.dart:931)
#5      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336)
#6      _DelayedData.perform (dart:async/stream_impl.dart:591)
#7      _StreamImplEvents.handleNext (dart:async/stream_impl.dart:707)
#8      _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:667)
#9      _rootRun (dart:async/zone.dart:1120)
#10     _CustomZone.run (dart:async/zone.dart:1021)
#11     _CustomZone.runGuarded (dart:async/zone.dart:923)
#12     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963)
#13     _rootRun (dart:async/zone.dart:1124)
#14     _CustomZone.run (dart:async/zone.dart:1021)
#15     _CustomZone.runGuarded (dart:async/zone.dart:923)
#16     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963)
#17     _microtaskLoop (dart:async/schedule_microtask.dart:41)
#18     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)
#19     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:116)
#20     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:173)
```

Otherwise the output looks good to me. For instance I do get back valid warnings like:

```bash
http://www-vbox.transloadit.com/blog/2011/05/fixing-amazon-s3/
- (704:0) '#s3-expo..' => http://www-vbox.transloadit.com/blog/2011/05/fixing-amazon-s3/#blog-posts (HTTP 200 but missing anchor)

http://www-vbox.transloadit.com/blog/2011/05/support-for-leaving-smaller-images-untouched/
- (629:24) 'full doc..' => http://www-vbox.transloadit.com/docs/transcoding/#image-manipulation-and-resizing (HTTP 200 but missing anchor)

http://www-vbox.transloadit.com/blog/2013/01/improvements-for-how-assembly-crashes-are-handled/
- (630:49) 'the new ..' => http://www-vbox.transloadit.com/accounts/api_settings (HTTP 500)
```

**Comment by kvz** [comment]
Running this build by the way, on latest Ubuntu: https://github.com/filiph/linkcheck/releases/download/v2.0.12/linkcheck-linux-x64.exe

**Comment by kvz** [comment]
Got one more:

```
INTERNAL ERROR: Sorry! Please open https://github.com/filiph/linkcheck/issues/new in your favorite browser and copy paste the following output there:


FileSystemException: writeFrom failed, path = '' (OS Error: Resource temporarily unavailable, errno = 11)

#0      _RandomAccessFile.writeFromSync (dart:io/file_impl.dart:879)

#1      _StdConsumer.addStream.<anonymous closure> (dart:io/stdio.dart:344)

#2      _rootRunUnary (dart:async/zone.dart:1132)

#3      _CustomZone.runUnary (dart:async/zone.dart:1029)

#4      _CustomZone.runUnaryGuarded (dart:async/zone.dart:931)

#5      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336)

#6      _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263)

#7      _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:764)

#8      _StreamController._add (dart:async/stream_controller.dart:640)

#9      _StreamController.add (dart:async/stream_controller.dart:586)

#10     _StreamSinkImpl.add (dart:io/io_sink.dart:156)

#11     _IOSinkImpl.write (dart:io/io_sink.dart:289)

#12     _IOSinkImpl.writeln (dart:io/io_sink.dart:309)

#13     _StdSink.writeln (dart:io/stdio.dart:341)

#14     printWithoutAnsi.print (package:linkcheck/src/writer_report.dart:159)

#15     printWithoutAnsi (package:linkcheck/src/writer_report.dart:166)

#16     reportForWriters (package:linkcheck/src/writer_report.dart:100)

#17     run (package:linkcheck/linkcheck.dart:276)

#18     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71)

#19     _rootRunUnary (dart:async/zone.dart:1132)

#20     _CustomZone.runUnary (dart:async/zone.dart:1029)

#21     _FutureListener.handleValue (dart:async/future_impl.dart:137)

#22     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678)

#23     Future._propagateToListeners (dart:async/future_impl.dart:707)

#24     Future._completeWithValue (dart:async/future_impl.dart:522)

#25     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:30)

#26     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:288)

#27     crawl (package:linkcheck/src/crawl.dart:0)

#28     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71)

#29     _rootRunUnary (dart:async/zone.dart:1132)

#30     _CustomZone.runUnary (dart:async/zone.dart:1029)

#31     _FutureListener.handleValue (dart:async/future_impl.dart:137)

#32     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678)

#33     Future._propagateToListeners (dart:async/future_impl.dart:707)

#34     Future._completeWithValue (dart:async/future_impl.dart:522)

#35     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:30)

#36     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:288)

#37     Pool.close (package:linkcheck/src/worker/pool.dart:0)

#38     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71)

#39     _rootRunUnary (dart:async/zone.dart:1132)

#40     _CustomZone.runUnary (dart:async/zone.dart:1029)

#41     _FutureListener.handleValue (dart:async/future_impl.dart:137)

#42     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678)

#43     Future._propagateToListeners (dart:async/future_impl.dart:707)

#44     Future._completeWithValue (dart:async/future_impl.dart:522)

#45     Future.wait.<anonymous closure> (dart:async/future.dart:400)

#46     _rootRunUnary (dart:async/zone.dart:1132)

#47     _CustomZone.runUnary (dart:async/zone.dart:1029)

#48     _FutureListener.handleValue (dart:async/future_impl.dart:137)

#49     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678)

#50     Future._propagateToListeners (dart:async/future_impl.dart:707)

#51     Future._completeWithValue (dart:async/future_impl.dart:522)

#52     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:552)

#53     _rootRun (dart:async/zone.dart:1124)

#54     _CustomZone.run (dart:async/zone.dart:1021)

#55     _CustomZone.runGuarded (dart:async/zone.dart:923)

#56     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963)

#57     _microtaskLoop (dart:async/schedule_microtask.dart:41)

#58     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)

#59     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:391)

#60     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:416)

#61     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172)

```

**Comment by filiph** [comment]
Ooof, this took way longer than I anticipated, but it's finally fixed in version `2.0.15`. If things don't work as expected, please run linkcheck with `--verbose` and paste the output here. Thanks for the patience!
